File: WMS.EntityCore/Propietario/clsBePropietario_reglas_enc.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations.Schema;
using WMS.EntityCore.Trans_re;
using WMSWebAPI.Be;

namespace WMS.EntityCore.Propietario
{
    public class clsBePropietario_reglas_enc : ICloneable
    {
        [Column("IdReglaPropietarioEnc")]
        [DisplayName("IdReglaPropietarioEnc")]
        public int IdReglaPropietarioEnc { get; set; } = 0;

        [Column("IdReglaRecepcion")]
        [DisplayName("IdReglaRecepcion")]
        public int IdReglaRecepcion { get; set; } = 0;

        [Column("IdPropietario")]
        [DisplayName("IdPropietario")]
        public int IdPropietario { get; set; } = 0;

        [Column("IdMensajeRegla")]
        [DisplayName("IdMensajeRegla")]
        public int IdMensajeRegla { get; set; } = 0;

        [Column("user_agr")]
        [DisplayName("user_agr")]
        public string User_agr { get; set; } = "";

        [Column("fec_agr")]
        [DisplayName("fec_agr")]
        public DateTime Fec_agr { get; set; } = DateTime.Now;

        [Column("user_mod")]
        [DisplayName("user_mod")]
        public string User_mod { get; set; } = "";

        [Column("fec_mod")]
        [DisplayName("fec_mod")]
        public DateTime Fec_mod { get; set; } = DateTime.Now;

        [Column("activo")]
        [DisplayName("activo")]
        public bool Activo { get; set; } = false;
        public bool IsNew { get; set; }
        public clsBeReglas_RecepcionRes Regla { get; set; } = new clsBeReglas_RecepcionRes();
        public string Propietario { get; set; } = "";
        public clsBeMensaje_regla Mensaje { get; set; } = new clsBeMensaje_regla();
        public List<clsBePropietario_reglas_det> ReglasDet { get; set; } = new List<clsBePropietario_reglas_det>();
        public clsBePropietario_reglas_enc() { }

        public object Clone()
        {
            return MemberwiseClone();
        }
    }
}

File: TOMIMSV4/Entity/Mantenimientos/Propietario/Reglas/clsBePropietario_reglas_enc.vb
Public Class clsBePropietario_reglas_enc
    Implements ICloneable
    Implements IDisposable

    Public Property IdReglaPropietarioEnc() As Integer = 0
    Public Property IdReglaRecepcion() As Integer = 0
    Public Property IdPropietario() As Integer = 0
    Public Property IdMensajeRegla() As Integer = 0
    Public Property User_agr() As String = ""
    Public Property Fec_agr() As Date = Date.Now
    Public Property User_mod() As String = ""
    Public Property Fec_mod() As Date = Date.Now
    Public Property Activo() As Boolean = False

    Public Property TipoRegla() As String = ""

    Sub New()
    End Sub

    Sub New(ByRef IdReglaPropietarioEnc As Integer, ByVal IdReglaRecepcion As Integer, ByVal IdPropietario As Integer, ByVal IdMensajeRegla As Integer, ByVal user_agr As String, ByVal fec_agr As Date, ByVal user_mod As String, ByVal fec_mod As Date, ByVal activo As Boolean)
        Me.IdReglaPropietarioEnc = IdReglaPropietarioEnc
        Me.IdReglaRecepcion = IdReglaRecepcion
        Me.IdPropietario = IdPropietario
        Me.IdMensajeRegla = IdMensajeRegla
        Me.User_agr = user_agr
        Me.Fec_agr = fec_agr
        Me.User_mod = user_mod
        Me.Fec_mod = fec_mod
        Me.Activo = activo
    End Sub
    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function
End Class
